pkg/ibmvpc/vpc.go

```diff
@@ -346,6 +368,23 @@ func (n *NACL) AllowedConnectivity(src, dst vpcmodel.Node, isIngress bool) (*com
 	return n.analyzer.AllowedConnectivity(subnetCidr, inSubnetCidr, targetNode.Cidr(), isIngress)
 }
 
+func (n *NACL) RulesInConnectivity(src, dst vpcmodel.Node, conn *common.ConnectionSet, isIngress bool) ([]int, error) {
+	targetNode, subnetCidr, inSubnetCidr, err := n.initConnectivityComputation(src, dst, isIngress)
+	if err != nil {
+		return nil, err
+	}
+	// check if the subnet of the given node is affected by this nacl
+	if _, ok := n.subnets[subnetCidr]; !ok {
+		return nil, nil // not affected by current nacl
+	}
+	// nacl has no control on traffic between two instances in its subnet; this is marked by a rule with index -1
+	// which is not printed but only signals that this filter does not block (since there are rules)
+	if allInSubnet, err := common.IsAddressInSubnet(targetNode.Cidr(), subnetCidr); err == nil && allInSubnet {
+		return []int{-1}, nil
```

Comment:
where do you check this value of `-1`?

Reply:
Excellent catch. Now it is checked in `func (na *NACLAnalyzer) StringRules`.

Reply:
can you add a `const` for `-1` ?

Reply:
done